compiler/noirc_evaluator/src/ssa/opt/inlining.rs

```diff
@@ -81,47 +89,47 @@ impl Function {
     pub(super) fn inlined(
         &self,
         ssa: &Ssa,
+        should_inline_call: &impl Fn(&PerFunctionContext, &Ssa, FunctionId) -> bool,
+    ) -> Function {
+        InlineContext::new(ssa, self.id()).inline_all(ssa, &should_inline_call)
+    }
+
+    /// Generic function that determines whether a function should inline a call.
+    pub(super) fn should_inline_call(
```

Comment:
We don't want to enshrine this as the only way of determining whether a function call should be inlined or not (quite the opposite) so I don't think it should be defined as a method on `Function`.